packages/epics/src/agreements/validation/agreement-form-payment-schedule-validation.ts

```diff
@@ -0,0 +1,47 @@
+import { isBefore } from 'date-fns';
+
+type DateRange = {
+  from: Date | undefined;
+  to: Date | undefined;
+};
+
+type Milestone = {
+  percentage: string;
+  dateRange: DateRange;
+};
+
+export const validateMilestones = (milestones: Milestone[]): true | string => {
+  const total = milestones.reduce(
+    (sum, m) => sum + Number(m.percentage || 0),
```

Comment:
I don't like this conversion. The data schema should be clear. Use a transformer within the UI component to convert from internal to display value.